Library/Homebrew/test.rb

```diff
@@ -25,7 +25,9 @@
 
   # tests can also return false to indicate failure
   Timeout.timeout TEST_TIMEOUT_SECONDS do
-    raise "test returned false" if formula.run_test == false
+    formula.with_logging("test") do
+      raise "test returned false" if formula.run_test == false
+    end
```

Comment:
I'm a bit surprised there's a `Formula#with_logging` invocation both here and in the `Formula#run_test` method called from here. Which one of these is redundant? Or are they both needed and why?